projects/Mallard/src/hooks/use-edition-provider.tsx

```diff
@@ -0,0 +1,260 @@
+import React, {
+    createContext,
+    useContext,
+    useEffect,
+    useState,
+    Dispatch,
+} from 'react'
+import { RegionalEdition, SpecialEdition } from 'src/common'
+import { eventEmitter } from 'src/helpers/event-emitter'
+import { defaultSettings } from 'src/helpers/settings/defaults'
+import {
+    defaultEditionCache,
+    selectedEditionCache,
+    editionsListCache,
+} from 'src/helpers/storage'
+import { errorService } from 'src/services/errors'
+import { defaultRegionalEditions } from '../../../Apps/common/src/editions-defaults'
+import NetInfo from '@react-native-community/netinfo'
+import { AppState, AppStateStatus } from 'react-native'
+import { remoteConfigService } from 'src/services/remote-config'
+import { locale } from 'src/helpers/locale'
+
+interface EditionsEndpoint {
+    regionalEditions: RegionalEdition[]
+    specialEditions: SpecialEdition[]
+}
+
+interface EditionState {
+    editionsList: EditionsEndpoint
+    selectedEdition: RegionalEdition | SpecialEdition
+    defaultEdition: RegionalEdition
+    storeSelectedEdition: (
+        chosenEdition: RegionalEdition | SpecialEdition,
+        type: 'RegionalEdition' | 'SpecialEdition' | 'TrainingEdition',
+    ) => void
+}
+
+export interface StoreSelectedEditionFunc {
+    (
+        chosenEdition: RegionalEdition | SpecialEdition,
+        type: 'RegionalEdition' | 'SpecialEdition' | 'TrainingEdition',
+    ): void
+}
+
+export const DEFAULT_EDITIONS_LIST = {
+    regionalEditions: defaultRegionalEditions,
+    specialEditions: [],
+}
+
+export const BASE_EDITION = defaultRegionalEditions[0]
+
+const localeToEdition = new Map<string, RegionalEdition>()
+localeToEdition.set('en_AU', defaultRegionalEditions[1])
+localeToEdition.set('en_US', defaultRegionalEditions[2])
+localeToEdition.set('en_GB', defaultRegionalEditions[0])
```

Comment:
Nit pick: arrange them in-order (0, 1, 2)

Also, I think it would make it more readable in here also in tests if we can read it using a key instead of index number (e.g. `defaultRegionalEditions['uk']`)

Reply:
I agree about readability here, its why I prefer objects. However i sacrificed it to maintain the required data structure for the endpoint. One alternative would be to create them as an object and then have an abstraction to get them into an array for use elsewhere. I will add that to my list of improvements